monitor/hosts.go

```diff
@@ -116,29 +115,25 @@ func terminateHosts(ctx context.Context, hosts []host.Host, settings *evergreen.
 			for hostToTerminate := range work {
 				event.LogMonitorOperation(hostToTerminate.Id, reason)
 
-				err := util.RunFunctionWithTimeout(func() error {
-					return terminateHost(ctx, hostToTerminate, settings)
-				}, 12*time.Minute)
-
-				if err != nil {
-					if strings.Contains(err.Error(), ec2.EC2ErrorNotFound) {
-						err = hostToTerminate.Terminate()
-						if err != nil {
-							catcher.Add(errors.Wrap(err, "unable to set host as terminated"))
-							continue
+				func() { // use a function so that the defer works
```

Comment:
What defer?

Reply:
line 121, if you defer in a for loop without wrapping it in a func, the defer exits when the goroutine exits, not when the loop ends (which is what we want in this case.)